lib/galaxy/files/sources/posix.py

```diff
@@ -105,14 +105,16 @@ def _write_from(
             target_native_path = os.path.normpath(target_native_path)
             assert target_native_path.startswith(os.path.normpath(effective_root))
 
-        target_native_path_parent = os.path.dirname(target_native_path)
+        target_native_path_parent, target_native_path_name = os.path.split(target_native_path)
         if not os.path.exists(target_native_path_parent):
             if self.allow_subdir_creation:
                 os.makedirs(target_native_path_parent)
             else:
                 raise Exception("Parent directory does not exist.")
 
-        shutil.copyfile(native_path, target_native_path)
+        target_native_path_part = os.path.join(target_native_path_parent, f"_{target_native_path_name}.part")
+        shutil.copyfile(native_path, target_native_path_part)
```

Comment:
Could you add a brief explanation why we first copy and then rename? I've read what you wrote in slack, but, I think, it will be helpful to have a clarifying line (note?) here in the code, otherwise the copyfile line might get cleverly refactored away down the road.

Reply:
Good idea!